src/Database/PostgreSQL/PQTypes/Model/Migration.hs

```diff
@@ -57,6 +59,26 @@ data MigrationAction m =
       (RawSQL ()) -- ^ Table name
       TableIndex  -- ^ Index
 
+  -- | Migration for modifying columns. Parameters are:
+  --
+  -- Name of the table that the cursor is associated with. It has to be the same as in the
+  -- cursor SQL, see the second parameter.
+  --
+  -- SQL that will be used for the cursor.
+  --
+  -- Function that takes a list of primary keys provided by the cursor SQL and
```

Comment:
Technically, it doesn't have to be primary keys right? As long as you can uniquely identify a column.

Reply:
unique not null, i suppose (just to be pedantic, but that's important)

Reply:
After re-reading my comment I have to make a correction: I mean **ROW**, not **COLUMN**.

But I'm not sure how `not null` applies here. You either uniquely identify a record or don't, there's no space for `NULL`s in that. Maybe you meant that column that one would/could reference has `UNIQUE NOT NULL` constraint, but I believe that is what **uniquely identify a row/record** implies in that case.

Reply:
unique not null is just the sql constraint. unique is a dangerous term when talking about sql databases, because `null = null` returns null, which is neither true nor false, except if you really are in a predicate context, where it finally becomes false. so i supposed you were talking about the constraint, which has to be unique not null if you really want to enforce the column to be unique (in the usual sense of unique). So as I said, that's me being pedantic, but i just wanted to be sure the constraint, in the end, is unique not null

Reply:
Isn't this what people mean when they say a candidate key

Reply:
Yeah more or less, i didn't want to confuse people. that could also work with superkeys, but those are relational algebra terms, like candidate keys. Anything that can identify uniquely a record. I just said unique not null, just because "unique" is tricky in SQL, I wanted to be sure that everybody was on the same page.